docs/ProposedOperators.md

```diff
@@ -0,0 +1,262 @@
+listed in the type string; their respective types must exactly match the corresponding types in the type string attribute.
+
+    tuple(type: string, values: ...) : tuple<T0,T1,...,Tn>
+    record(type: string, values: ...) : record<{name},{fieldNames},{T0,T1,...,Tn}>
+
+Accessing fields of tuples or records results in a type that depends on the field that is accessed. All field indices and names are static, so
+the type can be determined through ONNX type inference.
+
+    get(s: tuple<T0,T1,...,Tn>, i: int) : T[i]
+    get(s: record<{name},{fieldNames},{T0,T1,...,Tn}>, field: string) : T[{fieldName}]
+
+Record and tuple fields are modified in an immutable fashion, creating a new value with only some fields modified. Note the existence
+of operators to modify multiple fields in one fell swoop.
+
+    set(s: tuple<T0,T1,...,Tn>, i: int, value: T[i]) : tuple<T0,T1,...,Tn>
+    set(s: tuple<T0,T1,...,Tn>, i: sequence<int>, values: ...) : tuple<T0,T1,...,Tn>
+
+    set(s: record<{name},{fieldNames},{T0,T1,...,Tn}>, field: string, value: T[{fieldName}]) : record<{name},{fieldNames},{T0,T1,...,Tn}>
+    set(s: record<{name},{fieldNames},{T0,T1,...,Tn}>, field: sequence<string>, values: ...) : record<{name},{fieldNames},{T0,T1,...,Tn}>
+
+## Collections
+
+### Fold
+
+fold will accumulate a value, starting with the 'basis' argument, and then apply the function
+to the value and each element in the collection. The accumulate value is the result of the 
+operator. S and T are usually the same type, but do not have to be.
+
+    fold(input: tensor<T>,   basis: S, tfrm: (S,T)->S) : S
+    fold(input: sequence<T>, basis: S, tfrm: (S,T)->S) : S
+
+### Map
+
+map will produce a collection from a source collection and a function. 
+The output collection has the same shape as the input.
+
+    map(input: tensor<T>,   tfrm: T->U) : tensor<U>
+    map(input: sequence<T>, tfrm: T->U) : sequence<U>
+    map(input: map<S,T>,    tfrm: T->U) : map<S,U>
+
+### Scale
+
+scale will produce a collection from a source collection, a scale factor, and an offset.
+The output collection has the same shape as the input.
+
+    scale(input: tensor<T>,   scale: T, offset: T) : tensor<T>
+    scale(input: sequence<T>, scale: T, offset: T) : sequence<T>
+    scale(input: map<S,T>,    scale: T, offset: T) : map<S,T>
+
+### ZipMap
+
+zipmap will produce a collection from two source collections and a function. 
+The input shapes will be unified by the usual broadcasting semantics.
+The output collection has the same shape as the inputs.
+
+    zipmap(x: tensor<S>,   y: tensor<T>,   tfrm: (S,T)->U) : tensor<U>
+    zipmap(x: sequence<S>, y: sequence<T>, tfrm: (S,T)->U) : sequence<U>
+    zipmap(x: map<R,S>,    y: map<R,T>,    tfrm: (S,T)->U) : map<S,U>
+
+
+## Tensors
+
+### Dimensions
+
+Get the total number of elements, or the size of each dimension for a given tensor.
```

Comment:
As discussed earlier, I believe len, dims, transpose, and permute are already covered by existing ops. I suggest deleting these, or just document that these are already covered by existing ops.

Reply:
Okay, I removed them.